lib/std/heap.zig

```diff
@@ -33,7 +33,7 @@ fn cShrink(self: *Allocator, old_mem: []u8, old_align: u29, new_size: usize, new
 
 /// This allocator makes a syscall directly for every allocation and free.
 /// Thread-safe and lock-free.
-pub const direct_allocator = &direct_allocator_state;
+pub const direct_allocator = if (builtin.arch == .wasm32) wasm_allocator else &direct_allocator_state;
```

Comment:
Would this be better done on the `DirectAllocator` type itself?

Reply:
I'm not sure. Since the wasm allocator has global state, we need to ensure that direct_allocator and wasm_allocator refer to the same instance.